push.go

```diff
@@ -0,0 +1,67 @@
+package llblib
+
+import (
+	"strconv"
+
+	"github.com/distribution/reference"
+	"github.com/moby/buildkit/client"
+)
+
+type registryPushOpts struct {
+	attrs map[string]string
+}
+
+// RegistryPushOption can be used to modify a registry push request.
+type RegistryPushOption interface {
+	SetRegistryPushOption(*registryPushOpts)
+}
+
+type registryPushOptionFunc func(*registryPushOpts)
+
+func (f registryPushOptionFunc) SetRegistryPushOption(o *registryPushOpts) {
+	f(o)
+}
+
+// WithInsecurePush will allow pushing to an insecure registry.
+func WithInsecurePush() RegistryPushOption {
+	return registryPushOptionFunc(func(o *registryPushOpts) {
+		o.attrs["registry.insecure"] = "true"
+	})
+}
+
+// WithPushByDigest will push the image by digest.
+func WithPushByDigest() RegistryPushOption {
+	return registryPushOptionFunc(func(o *registryPushOpts) {
+		o.attrs["push-by-digest"] = "true"
+	})
+}
+
+// WithCompression will set the compression type for the push.
+func WithCompression(compression string, force bool) RegistryPushOption {
+	return registryPushOptionFunc(func(o *registryPushOpts) {
+		o.attrs["compression"] = compression
+		o.attrs["force-compression"] = strconv.FormatBool(force)
+		if compression == "zstd" || compression == "estargz" {
+			o.attrs["oci-mediatypes"] = "true"
```

Comment:
Not sure if we want to hardcode this in llblib. I think I added this to newt in order to support buildah (it is very fussy about media types, and the developers insist zstd isn't properly spec'd outside the OCI ecosystem).

I don't think it will break anything, but it may be unexpected behavior.

Reply:
I think it is necessary, at least it is also [hard-coded in hlb](https://github.com/openllb/hlb/blob/master/solver/solve.go#L236) for estargz, so probably similar for zstd